api/objects.proto

```diff
@@ -267,3 +268,50 @@ message Secret {
 	// Whether the secret is an internal secret (not set by a user) or not.
 	bool internal = 4;
 }
+
+// Resource is a top-level object with externally defined content and indexing.
+// SwarmKit can serve as a store for these objects without understanding their
+// meanings.
+message Resource {
```

Comment:
`Resource` in the context of Swarmkit generally refers to cluster resource as in resource management. This is a general object. How about naming it `Record`?